src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -81,6 +82,8 @@ func NewEncoder(
 	}
 }
 
+func (enc *encoder) SetSchema(descr namespace.SchemaDescr) {}
```

Comment:
It will get called on m3tsz encoder. The caller (in this case, buffer bucket) does not know what kind of encoder it is, it always call SetSchema right before Encode to support mid stream schema change.